File: 25-10-2024_Alternative_sorting.py
class Solution:
    def alternateSort(self, arr):
        arr.sort()
        n = len(arr)
        
        l, r = 0, n - 1
        ans = []
        flag = True
        
        for i in range(n):
            if flag:
                ans.append(arr[r])
                r -= 1
            else:
                ans.append(arr[l])
                l += 1
            flag = not flag
        
        return ans

File: 25-10-2024_Alternative_sorting.java
public static ArrayList<Integer> alternateSort(int[] arr) {
      Arrays.sort(arr);
      int n = arr.length;
      
      int l = 0, r = n - 1;
      ArrayList<Integer> ans = new ArrayList<>();
      boolean flag = true;
      
      for (int i = 0; i < n; i++) {
          if (flag) {
              ans.add(arr[r--]);
          } else {
              ans.add(arr[l++]);
          }
          flag = !flag;
      }
      
      return ans;
  }
}
